File: easy-sept-21-2019/solution.js
const fs = require("fs");
const path = require("path");
const filePath = path.join(__dirname, "input.txt");

const file = fs.readFileSync(filePath, { encoding: "utf-8" });
fileLines = file.split("\n");
numbers = fileLines[0].split(" ").map(value => parseInt(value));
targetSum = parseInt(fileLines[1]);

const result = (numbers, targetSum) => {
  const difference = new Set();
  for (var i = 0; i < numbers.length; i++) {
    if (difference.has(numbers[i])) {
      return true;
    }
    difference.add(targetSum - numbers[i]);
  }
  return false;
};

console.log(
  `Any 2 numbers from list add up to k: ${result(numbers, targetSum)}`
);

File: easy-sept-21-2019/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func readInputFromFile() ([]int, int) {
	file, _ := os.Open("./input.txt")
	defer file.Close()

	var numbers []int
	var targetSum int

	scanner := bufio.NewScanner(file)
	scanner.Scan()
	for _, numberStr := range strings.Split(scanner.Text(), " ") {
		number, _ := strconv.Atoi(numberStr)
		numbers = append(numbers, number)
	}
	scanner.Scan()
	targetSum, _ = strconv.Atoi(scanner.Text())

	return numbers, targetSum
}

func result(numbers []int, targetSum int) bool {
	difference := map[int]bool{}

	for _, number := range numbers {
		if _, ok := difference[number]; ok {
			return true
		}
		difference[targetSum-number] = true
	}

	return false
}

func main() {
	numbers, targetSum := readInputFromFile()
	fmt.Printf("Any 2 numbers from list add up to k: %t\n", result(numbers, targetSum))
}
